drools-ansible-rulebook-integration-protoextractor/src/main/java/org/drools/ansible/rulebook/integration/protoextractor/prototype/ExtractorPrototypeExpression.java

```diff
@@ -0,0 +1,50 @@
+package org.drools.ansible.rulebook.integration.protoextractor.prototype;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.drools.ansible.rulebook.integration.protoextractor.ExtractorUtils;
+import org.drools.ansible.rulebook.integration.protoextractor.ast.ExtractorNode;
+import org.drools.base.facttemplates.Fact;
+import org.drools.model.Prototype;
+import org.drools.model.PrototypeExpression;
+import org.drools.model.PrototypeFact;
+import org.drools.model.functions.Function1;
+
+public class ExtractorPrototypeExpression implements PrototypeExpression {
+    protected final ExtractorNode extractorNode;
+    protected final String computedFieldName;
+    protected final Collection<String> computedImpactedFields;
+
+    public ExtractorPrototypeExpression(ExtractorNode extractorNode) {
+        this.extractorNode = extractorNode;
+        this.computedFieldName = ExtractorUtils.getParts(extractorNode).stream().collect(Collectors.joining());
+        this.computedImpactedFields = Collections.singletonList(ExtractorUtils.getParts(extractorNode).get(0));
+    }
+
+    @Override
+    public Function1<PrototypeFact, Object> asFunction(Prototype prototype) {
+        return pf -> {
+            Map<String, Object> asMap = ((Fact) pf).asMap();
+            Object value = new ValueExtractionVisitor(asMap).visit(extractorNode);
+            return value;
+        };
+    }
+
+    // TODO used for indexing, normalizing chunks.
+    public String getFieldName() {
```

Comment:
Unfortunately this is breaking indexing, see https://github.com/kiegroup/drools/blob/main/drools-model/drools-canonical-model/src/main/java/org/drools/model/PrototypeDSL.java#L135 What I suggest is introducing an interface named something like `FieldAccessingExpression` declaring a `getFieldName()` method, making both this class and `PrototypeFieldValue` to implement it and changing the downcasts in `PrototypeDSL` to use it. The only problem is that this will of course require also a change in Drools, but this seems unavoidable (or at least I don't have any better idea).

Reply:
Also it would be nice to add a few assertions in some tests, for instance [here](https://github.com/kiegroup/drools-ansible-rulebook-integration/blob/main/drools-ansible-rulebook-integration-api/src/test/java/org/drools/ansible/rulebook/integration/api/AlphaTest.java#L13), to make sure that the alpha node is indexed as expected.

Reply:
seems no other way indeed given `PrototypeFieldValue` cannot be instantiated (due to constructor visibility)

Reply:
done with 
- https://github.com/kiegroup/drools/pull/5333
- https://github.com/kiegroup/drools-ansible-rulebook-integration/pull/56/commits/b2602bb2a23dd0701719073502824225802199e4

Reply:
What about just adding the extractor method to the `PrototypeExpression` interface returning an `Optional`? Like
`Optional<String> getIndexingKey();`

In this way we can avoid the instanceof and the downcasting: each expression can be indexable returning a value or not with an empty option

Reply:
thanks for the comment @danielezonca 
I believe this pends a confirmation from Mario these 2 guards:

- https://github.com/kiegroup/drools/pull/5333/files#diff-3f9ca0220c837c33d5a557b4af7727094a6c21475c926ce19dfc1bffd02c3c90R135
- https://github.com/kiegroup/drools/pull/5333/files#diff-3f9ca0220c837c33d5a557b4af7727094a6c21475c926ce19dfc1bffd02c3c90R186

are _semantically_ guarding JUST-and-only the indexing.

Reply:
@danielezonca I don't see any semantic difference between the method returning an Optional and the new interface and actually I still prefer to use the second to have clearer marker between which expression is indexable and which isn't.

Reply:
implemented https://github.com/kiegroup/drools-ansible-rulebook-integration/pull/56#discussion_r1236469355

with:
 - https://github.com/kiegroup/drools/pull/5333/commits/a26fce129055244517e2c94b5b239e53f318047c
 - https://github.com/kiegroup/drools-ansible-rulebook-integration/pull/56/commits/4dc46563be9b8807ebfd5b97cf148ef6afba6cd2